### DISCUSSION TITLE
Pelican v4.8.0 outputs duplicate content into output directory root

### DISCUSSION THREAD

**Discussion by mlindgren** [getpelican/pelican#3179] [Q&A] [answered]
I have a blog built on Pelican that I haven't been actively maintaining since 2017. I recently had a reason to update it again, so I started by installing the latest Pelican version. After regenerating my site, I noticed that all of the output is being duplicated: each article gets written once into my `ARTICLE_SAVE_AS` path as expected, and then again into `output/{slug}.html`. For example, for one article I get these two outputs:

- `output/entry/2015/12/27/early-access-roundup/index.html`
- `output/early-access-roundup.html`

I don't want the second one, as I don't like how it clutters up the output directory. Is this standard Pelican behavior now, or is it a bug? If the former, is there a way to disable it?

In case it helps, my complete `pelicanconf.py` is [here](https://github.com/mlindgren/blog/blob/master/pelicanconf.py); this repo also contains the rest of the content for the site.

**Comment by justinmayer** [comment]
I don't see anything in your settings file that would cause that. I suggest running `pelican-quickstart` to create a new site. Then copy over your content. Then edit the generated settings file and only change the `ARTICLE_URL` and `ARTICLE_SAVE_AS` settings to the desired patterns. Then generate the site and see if you see the same duplicate behavior described above.

**Comment by avaris** [comment, marked as answer]
Are you sure you're not looking at the output of multiple pelican runs with different configurations? I'd suggest clearing your `output` folder and checking again: either manually or via running `pelican` with `--delete-output-directory` option.

**Reply by mlindgren** [reply]
This was it, thanks! I'm a bit embarrassed that this hadn't occurred to me. I guess the first time I ran Pelican after upgrading, something about my old config file caused this issue, and then after fixing that I didn't clean the output directory before subsequent runs.

**Reply by mlindgren** [reply]
@avaris By the way, I didn't want to overcomplicate this issue, but it seems relevant now: part of the reason I had trouble root-causing this is because Pelican's logging is not printing format strings correctly; the `%s` values are not replaced. e.g.

```
INFO: Writing %s
INFO: Writing %s
INFO: Creating directory %s
INFO: Creating directory %s
INFO: Copying %s to %s
INFO: Copying %s to %s
```

Any idea what's causing that?

**Reply by avaris** [reply]
No, that's not normal. If you can describe a way to reproduce that I'll look into it.
